astroplan/core.py

```diff
@@ -22,109 +38,152 @@
-            The ambient pressure.
+        self.name = name
+        self.pressure = pressure
+        self.temperature = temperature
+        self.relative_humidity = relative_humidity
 
-        relative_humidity : `~astropy.units.Quantity`
-            The ambient relative humidity.
+        # If lat/long given instead of EarthLocation, convert them
+        # to EarthLocation
+        if location is None and (latitude is not None and
+                                 longitude is not None):
+            self.location = EarthLocation.from_geodetic(longitude, latitude,
+                                                        elevation)
 
-        temperature :
-            The ambient temperature.
-        """
-        raise NotImplementedError()
+        elif isinstance(location, EarthLocation):
+            self.location = location
 
-    def get_date(self, input_date_time, timezone=None):
-        """
-        Builds an object containing date and time information.
+        else:
+            raise TypeError('Observatory location must be specified with '
+                            'either (1) an instance of '
+                            'astropy.coordinates.EarthLocation or (2) '
+                            'latitude and longitude in degrees as '
+                            'accepted by astropy.coordinates.Latitude and '
+                            'astropy.coordinates.Latitude.')
 
-        Default time zone is UTC.
-        If time zone (local or otherwise) requested, then uses `datetime.datetime`
-        and pytz.timezone to convert from UTC.
+        # Accept various timezone inputs, default to UTC
+        if isinstance(timezone, datetime.tzinfo):
+            self.timezone = timezone
+        elif isinstance(timezone, str) or isinstance(timezone, unicode):
+            self.timezone = pytz.timezone(timezone)
+        else:
+            raise TypeError('timezone keyword should be a string, or an '
+                            'instance of datetime.tzinfo')
 
-        Must first create date_time object with get_date() before using any other 
-        Observer methods, all of which require one as an argument.
+    def altaz(self, time, target=None, obswl=None):
+        """
+        Calculates an altitude/azimuth frame (if `target` not specified)
+        or coordinate (otherwise).
 
         Parameters
         ----------
-        input_date_time : WHAT TYPE IS astropy.time OBJECT?
+        time : `~astropy.time.Time`
+            Astropy time object.
+
+        target : `~astroplan.FixedTarget`, `~astropy.coordinates.SkyCoord`, defaults to `None` (optional)
+            Celestial object of interest. If `target` is `None`, return the
+            `~astropy.coordinates.AltAz` frame without coordinates.
+
+        obswl : `~astropy.units.Quantity` (optional)
+            Wavelength of the observation used in the calculation.
+
+        Returns
+        -------
+        `~astropy.coordinates.AltAz`
+            Frame or coordinate.
```

Comment:
Isn't it either an `AltAz` *or* a `SkyCoord?`.  More properly (and this is a subtle but important point), if `target` is not None, this is the type that `target` was.  That is, if `target` is a `SkyCoord`, `tranform_to` yields a new `SkyCoord`.  But if it's, say, an `ICRS` coordinate object, `tranform_to` will yield an `AltAz` coordinate.  So this has to be gotten across somehow in the "Returns" description.